core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -62,6 +66,15 @@ public BlockBuilder getFieldBlockBuilder(int fieldIndex)
         return fieldBlockBuilders[fieldIndex];
     }
 
+    public void reset(int rowIndex)
+    {
+        this.rowIndex = rowIndex;
+        this.currentFieldIndexToWrite = 0;
+        this.initialBlockBuilderSize = computeInitialBlockBuilderSize();
+        this.positionsWritten = 0;
```

Comment:
I don't think we need separate `positionsWritten` field

Reply:
ok